lib/SILOptimizer/Transforms/CopyForwarding.cpp

```diff
@@ -341,6 +341,7 @@ class AnalyzeForwardUse
     case SILArgumentConvention::Indirect_In:
       return true;
     case SILArgumentConvention::Indirect_In_Guaranteed:
+    case SILArgumentConvention::Indirect_In_CXX:
```

Comment:
Just to help me better understand this patch, because I don't see this case being defined in this patch, and the `default` case is `llvm_unreachable`.

Does this mean that the `Indirect_In_Cxx` case was not possible prior to this patch? (I'm guessing it's the additions to `lib/SIL/IR/SILFunctionType.cpp` that make this case possible?)

Reply:
I hit the unreachable assertion with the test case I added. My suspicion is that it was possible to trigger this even before my patch we just did not run into it with an assertions enabled compiler.